### PR TITLE
[pmon daemon check] refactoring pmon daemon state check code

### PR DESCRIPTION
**Pull request by yxieca**
Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you do it?
- Move daemon state checking code to test/common
- Get all running daemon state from pmon docker, and filter out exemptions and disabled demons.
- Change state checking code to validate returned states.

#### How did you verify/test it?
This feature is currently only used in reboot tests:
============================================================================================= test session starts =============================================================================================
platform linux2 -- Python 2.7.12, pytest-4.6.8, py-1.8.1, pluggy-0.13.1
ansible: 2.8.7
rootdir: /var/src/sonic-mgmt/tests, inifile: pytest.ini
plugins: ansible-2.2.2
collected 7 items                                                                                                                                                                                             

platform/test_reboot.py::test_cold_reboot PASSED                                                                                                                                                        [ 14%]
platform/test_reboot.py::test_fast_reboot PASSED                                                                                                                                                        [ 28%]
platform/test_reboot.py::test_warm_reboot PASSED                                                                                                                                                        [ 42%]
platform/test_reboot.py::test_power_off_reboot[15] SKIPPED                                                                                                                                              [ 57%]
platform/test_reboot.py::test_power_off_reboot[5] SKIPPED                                                                                                                                               [ 71%]
platform/test_reboot.py::test_watchdog_reboot SKIPPED                                                                                                                                                   [ 85%]
platform/test_reboot.py::test_continuous_reboot PASSED                                                                                                                                                  [100%]

Tested:
- No pmon_daemon_control.json
- pmon_daemon_control.json with daemon enabled
- pmon_daemon_control.json with daemon disabled

### CODE REVIEW

File: tests/common/devices.py [hunk 1]
```diff
@@ -248,20 +254,26 @@ def get_pmon_daemon_list(self):
             logging.debug("Original file content is %s" % str(json_data))
             for key in full_daemon_tup:
```
Comment (neethajohn) [inline tests/common/devices.py:255]: where is full_deamon_tup defined?

File: tests/common/devices.py [hunk 2]
```diff
@@ -232,14 +232,20 @@ def get_crm_resources(self):
...
         """
-        full_daemon_tup = ('xcvrd', 'ledd', 'psud', 'syseepromd')
+        # some services are meant to have a short life span or not part of the daemons
+        exemptions = ('lm-sensors', 'start.sh', 'rsyslogd')
+
+        daemons = self.shell('docker exec pmon supervisorctl status')['stdout_lines']
+
         daemon_ctl_key_prefix = 'skip_'
-        daemon_list = []
         daemon_config_file_path = os.path.join('/usr/share/sonic/device', self.facts["platform"], 'pmon_daemon_control.json')
```
Comment (neethajohn) [inline tests/common/devices.py:249]: don't you need hwsku info? the path specified in the summary

Comment (jleveque) [reply tests/common/devices.py:249]: `pmon_daemon_control.json` is platform-specific, and is shared among that SKUs for that platform. Thus, it is under the platform directory (`self.facts["platform"]`).

File: tests/common/devices.py (hunk 1 above)
Comment (jleveque) [reply tests/common/devices.py:255]: Good catch! It was removed above.

File: tests/common/devices.py [hunk 3]
```diff
@@ -232,14 +232,20 @@ def get_crm_resources(self):
 
         return result
 
-    def get_pmon_daemon_list(self):
+    def get_pmon_daemon_states(self):
         """
-        @summary: get pmon daemon list from the config file (/usr/share/sonic/device/{platform}/{hwsku}/pmon_daemon_control.json)
+        @summary: get state list of daemons from pmon docker.
+                  Referencing (/usr/share/sonic/device/{platform}/{hwsku}/pmon_daemon_control.json)
```
Comment (daall) [inline tests/common/devices.py:238]: Per below discussion:
```suggestion
                  Referencing (/usr/share/sonic/device/{platform}/pmon_daemon_control.json)
```

File: tests/common/devices.py (hunk 1 above)
Comment (yxieca) [reply tests/common/devices.py]: Thanks for catching this. Fixed now!

File: tests/common/devices.py (hunk 3 above)
Comment (yxieca) [reply tests/common/devices.py]: Thanks! Fixed now!

**Pull request merged**